incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/OltpExporterHandler.java

```diff
@@ -104,17 +112,30 @@ public int export()
         long now = System.currentTimeMillis();
         if (now >= nextAttempt)
         {
-            if (response == null || response.isDone())
+            if (signals.contains(METRICS) && (metricsResponse == null || metricsResponse.isDone()))
             {
-                String json = serializer.serializeAll();
-                HttpRequest request = HttpRequest.newBuilder()
+                String metricsJson = metricsSerializer.serializeAll();
+                HttpRequest metricsRequest = HttpRequest.newBuilder()
                     .uri(metricsEndpoint)
                     .header("Content-Type", "application/json")
-                    .POST(HttpRequest.BodyPublishers.ofString(json))
+                    .POST(HttpRequest.BodyPublishers.ofString(metricsJson))
                     .timeout(timeoutInterval)
                     .build();
-                response = httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString());
-                response.thenAccept(responseHandler);
+                metricsResponse = httpClient.sendAsync(metricsRequest, HttpResponse.BodyHandlers.ofString());
+                metricsResponse.thenAccept(responseHandler);
+                nextAttempt = now + retryInterval;
+            }
+            if (signals.contains(LOGS) && (logsResponse == null || logsResponse.isDone()))
+            {
+                String logsJson = logsSerializer.serializeAll();
+                HttpRequest logsRequest = HttpRequest.newBuilder()
+                    .uri(logsEndpoint)
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(logsJson))
+                    .timeout(timeoutInterval)
+                    .build();
+                logsResponse = httpClient.sendAsync(logsRequest, HttpResponse.BodyHandlers.ofString());
+                logsResponse.thenAccept(responseHandler);
```

Comment:
Let's split this into `int exportMetrics()` and `int exportLogs()` called from `int export()`.

Reply:
done